### PR TITLE
DPTS-584 Symptom group discriminators and synonym lambdas

### PR DESCRIPTION
**Pull request by tamaragoldschmidt**
Symptom group discriminators and symptom discriminator synonyms house keeping lambdas and general refactoring

### CODE REVIEW

File: application/hk/symptomdiscriminatorsynonyms/handler.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    for row_number, row_values in row_data.items():
+        try:
+            record_exists = does_sds_record_exist(db_connection, row_values, event["env"])
+            if common.valid_action(record_exists, row_values, event["env"], "UPDATE"):
+                query, data = generate_db_query(row_values, event["env"])
+                database.execute_db_query(
+                    db_connection, query, data, row_number, row_values, summary_count_dict, event["env"]
+                )
+            else:
+                print("here")
```
Comment (timrickwood) [inline application/hk/symptomdiscriminatorsynonyms/handler.py:96]: Either remove print instruction or replace with logger for audit if something useful to be gained here

File: build/automation/var/profile/live.mk [hunk 2]
```diff
@@ -11,7 +11,7 @@ SERVICE_PREFIX := $(PROJECT_ID)-$(ENV)
 # Infrastructure variables
 
 STACKS := secrets,s3,security-groups
-TASKS := filter,referralroles,symptomgroups,symptomdiscriminators,servicetypes
+TASKS := filter,referralroles,symptomgroups,symptomdiscriminators,servicetypes,symptomdiscriminatorsynonyms
```
Comment (timrickwood) [inline build/automation/var/profile/live.mk:14]: should this also have symptomgroupdiscriminators added

File: build/automation/var/profile/put.mk [hunk 3]
```diff
@@ -11,7 +11,7 @@ SERVICE_PREFIX := $(PROJECT_ID)-$(ENV)
 # Infrastructure variables
 
 STACKS := secrets,s3,security-groups
-TASKS := filter,referralroles,symptomgroups,symptomdiscriminators,servicetypes
+TASKS := filter,referralroles,symptomgroups,symptomdiscriminators,servicetypes,symptomdiscriminatorsynonyms
```
Comment (timrickwood) [inline build/automation/var/profile/put.mk:14]: should this also have symptomgroupdiscriminators added?

File: build/jenkins/Jenkinsfile.prod [hunk 4]
```diff
@@ -1,7 +1,7 @@
 pipeline {
   agent any
   parameters {
-    choice(name: 'Task', choices: ['none', 'all', 'filter', 'referralroles','symptomgroups','symptomdiscriminators'], description: 'Select which task to deploy')
+    choice(name: 'Task', choices: ['none', 'all', 'filter', 'referralroles','symptomgroups','symptomdiscriminators','symptomdiscriminatorsynonyms'], description: 'Select which task to deploy')
```
Comment (timrickwood) [inline build/jenkins/Jenkinsfile.prod:4]: should this also have symptomgroupdiscriminators added

File: test/unit/utilities/test_utilities_common.py [hunk 5]
```diff
@@ -38,6 +38,14 @@ def test_invalid_create_action():
     csv_dict["action"] = "CREATE"
     assert not common.valid_action(True,csv_dict, 'test')
 
+# def test_invalid_create_action():
```
Comment (timrickwood) [inline test/unit/utilities/test_utilities_common.py:41]: Why have we removed this one? Invalid test? Superfluous? Replaced? If it is not needed then remove

File: application/hk/symptomdiscriminatorsynonyms/handler.py (hunk 1 above)
Comment (tamaragoldschmidt) [reply application/hk/symptomdiscriminatorsynonyms/handler.py:96]: removed - debug comment left behind

File: build/automation/var/profile/live.mk (hunk 2 above)
Comment (tamaragoldschmidt) [reply build/automation/var/profile/live.mk:14]: done

File: build/automation/var/profile/put.mk (hunk 3 above)
Comment (tamaragoldschmidt) [reply build/automation/var/profile/put.mk:14]: done

File: build/jenkins/Jenkinsfile.prod (hunk 4 above)
Comment (tamaragoldschmidt) [reply build/jenkins/Jenkinsfile.prod:4]: done

File: test/unit/utilities/test_utilities_common.py (hunk 5 above)
Comment (tamaragoldschmidt) [reply test/unit/utilities/test_utilities_common.py]: uncommented and changed title

**Review by timrickwood — APPROVED** [review]

**Pull request merged**